### PR TITLE
vis 0.2 (new formula)

### PR DESCRIPTION
**Pull request by twe4ked**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### New Formulae Submissions:

- [x] Does your submission pass `brew audit --strict --online vis`?
- [x] Have you built your formula locally prior to submission with `brew install vis`?

### CODE REVIEW

File: Library/Formula/vis.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "vis"
```
Comment (bfontaine) [inline Library/Formula/vis.rb]: Please use the full path: `"#{bin}/vis"`.

File: Library/Formula/vis.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Vis < Formula
+  desc "Aims to be a modern, legacy free, simple yet efficient vim-like editor."
```
Comment (bfontaine) [inline Library/Formula/vis.rb]: You might want to shorten this to `"Modern, legacy free, simple yet efficient vim-like editor"`.

**Comment by bfontaine** [comment]
It’ll need to be `keg_only` because it conflicts with `vis(1)`, as you wrote on https://github.com/martanne/vis/issues/101#issuecomment-203849635.

**Comment by twe4ked** [comment]
Thanks for the review @bfontaine!

Force pushed to keep it a single commit. Here is the diff for review:

``` diff
diff --git a/Library/Formula/vis.rb b/Library/Formula/vis.rb
index 508425e..1bd52a6 100644
--- a/Library/Formula/vis.rb
+++ b/Library/Formula/vis.rb
@@ -1,10 +1,12 @@
 class Vis < Formula
-  desc "Aims to be a modern, legacy free, simple yet efficient vim-like editor."
+  desc "Modern, legacy free, simple yet efficient vim-like editor"
   homepage "http://www.brain-dump.org/projects/vis"
   url "http://www.brain-dump.org/projects/vis/vis-0.2.tar.gz"
   sha256 "3e5b81d760849c56ee378421e9ba0f653c641bf78e7594f71d85357be99a752d"
   head "https://github.com/martanne/vis.git"

+  keg_only :provided_by_osx
+
   depends_on "libtermkey"
   depends_on "lua"
   depends_on "lpeg" => :lua
@@ -18,6 +20,6 @@ class Vis < Formula
   end

   test do
-    system "vis"
+    system "#{bin}/vis"
   end
 end
```

File: Library/Formula/vis.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class Vis < Formula
+  desc "Modern, legacy free, simple yet efficient vim-like editor"
+  homepage "http://www.brain-dump.org/projects/vis"
+  url "http://www.brain-dump.org/projects/vis/vis-0.2.tar.gz"
+  sha256 "3e5b81d760849c56ee378421e9ba0f653c641bf78e7594f71d85357be99a752d"
+  head "https://github.com/martanne/vis.git"
+
+  keg_only :provided_by_osx
```
Comment (bfontaine) [inline Library/Formula/vis.rb]: Technically it’s not provided by OS X because it’s not the same tool. Something like ``keg_only "Shadows OS X' `vis` command"`` should be more appropriate. See e.g. [`unixodbc.rb`](https://github.com/Homebrew/homebrew/blob/bf20a138788ae2355052619ab59a9b23c3b8e10a/Library/Formula/unixodbc.rb#L21) and [`e2fsprogs.rb`](https://github.com/Homebrew/homebrew/blob/bf20a138788ae2355052619ab59a9b23c3b8e10a/Library/Formula/e2fsprogs.rb#L16).

**Comment by twe4ked** [comment]
Oh, I misunderstood.

```
diff --git a/Library/Formula/vis.rb b/Library/Formula/vis.rb
index 1bd52a6..cb84506 100644
--- a/Library/Formula/vis.rb
+++ b/Library/Formula/vis.rb
@@ -5,7 +5,7 @@ class Vis < Formula
   sha256 "3e5b81d760849c56ee378421e9ba0f653c641bf78e7594f71d85357be99a752d"
   head "https://github.com/martanne/vis.git"

-  keg_only :provided_by_osx
+  keg_only "Shadows OS X' `vis` command"

   depends_on "libtermkey"
   depends_on "lua"
```

File: Library/Formula/vis.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+class Vis < Formula
+  desc "Modern, legacy free, simple yet efficient vim-like editor"
+  homepage "http://www.brain-dump.org/projects/vis"
+  url "http://www.brain-dump.org/projects/vis/vis-0.2.tar.gz"
+  sha256 "3e5b81d760849c56ee378421e9ba0f653c641bf78e7594f71d85357be99a752d"
+  head "https://github.com/martanne/vis.git"
+
+  keg_only "Shadows OS X' `vis` command"
```
Comment (mikemcquaid) [inline Library/Formula/vis.rb]: `OS X's`

File: Library/Formula/vis.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  depends_on "libtermkey"
+  depends_on "lua"
+  depends_on "lpeg" => :lua
```
Comment (mikemcquaid) [inline Library/Formula/vis.rb]: This needs to be vendored in the formula; we don't really support language dependencies like this in core any more due to the crappy user experience.

Comment (twe4ked) [inline Library/Formula/vis.rb]: Is there an example of something similar in another formula?

File: Library/Formula/vis.rb (hunk 4 above)
Comment (twe4ked) [inline Library/Formula/vis.rb]: :flushed:

File: Library/Formula/vis.rb (hunk 5 above)
Comment (mikemcquaid) [inline Library/Formula/vis.rb]: Yeh but I don't believe for Lua. See `ansible.rb` for a Python example.

**Comment by twe4ked** [comment]
Looks like this is being tackled by someone else https://github.com/Homebrew/homebrew-core/pull/127. Should I close this one?

**Comment by mikemcquaid** [comment]
@twe4ked Yeh, may as well, thanks.
